design/API.md

```diff
@@ -276,3 +276,268 @@ Get the status of autotune.
 ```
 Healthy
 ```
+
+##  createExperiment
+Launch an experiment for a particular deployment with a config recommended by the Recommendation Manager 
+
+**Request**
+`GET /createExperiment`
+
+`curl -H 'Accept: application/json' -d <INPUT JSON> http://<URL>:<PORT>/createExperiment`
+
+*Description:*
+
+The API endpoint `createExperiment` expects the data object of type `JSON` with a structure as shown
+below. The Input JSON holds details of the configuration to try, settings to that particular deployment
+and the details of the deployment.
+
+*Sections:* 
+    
+    - Metadata
+    - Info
+    - Settings
+    - Deployments
+
+Metadata section consists of params like `experiment_id`, `name` etc which are top level info points for the experiment
+
+Info section consists of the trial information details like `trial_id`, `trial_num`
+
+Settings section consists of 2 sub-sections 
+
+
+    - trial_settings
+    - deployment_settings
+
+
+Trial Settings consists of setting related to trial like 
+
+`total_duration` : How long a trial experiment should run for ?
+`warmup_cycles` : Number of iterations for waiting for warmup
+`warmup_duration` : Time to wait for each iteration
+`measurement_cycles` : Number of iterations for measuring the metrics
+`measurement_duration` : Time to collect metrics in each iteration
+
+Deployment Settings consists of settings related to deployment like
+
+`deployment_info` : Metadata about the deployment like name of the deployment (will be ignored and warned if doesn't match in case of rollingUpdate), target environment (dev / qa / prod) 
+`deployment_policy` : What's the type of policy ? (newDeployment / rollingUpdate)
+`deployment_tracking` : Which deployments to track (either prod or training or both)
+
+Deployments section consists of the deployments to track and also their respective configs
+
+`deployment_name` : This is name of the target deployment to apply the config.
+
+*Sample Input JSON:*
+
+```
+{
+  "experiment_id": "2190310A384BC90EF",
+  "name": "petclinic-autotune",
+  "info": {
+    "trial_id": "",
+    "trial_num": 1
+  },
+  "settings": {
+    "trial_settings": {
+      "total_duration": "20 mins",
+      "warmup_cycles": 5,
+      "warmup_duration": "1 min",
+      "measurement_cycles": 15,
+      "measurement_duration": "1 min"
+    },
+    "deployment_settings": {
+      "deployment_info": {
+        "deployment_name" : "petclinic-sample",
+        "target_env" : "qa"
+      },
+      "deployment_policy" : {
+        "type" : "rollingUpdate"
+      },
+      "deployment_tracking": {
+        "trackers": [
+          "training",
+          "production"
+        ]
+      }
+    }
+  },
+  "deployments": [
+    {
+      "type" : "training",
+      "deployment_name": "petclinic-sample",
+      "namespace" : "default",
+      "state": "",
```

Comment:
We need to remove state and result fields from input json as it will be part of output json from EM